site/content/en/latest/install/custom-cert.md

```diff
@@ -0,0 +1,146 @@
+   apiVersion: cert-manager.io/v1
+   kind: Issuer
+   metadata:
+     labels:
+       app.kubernetes.io/name: envoy-gateway
+     name: selfsigned-issuer
+     namespace: envoy-gateway-system
+   spec:
+     selfSigned: {}
+   ---
+   apiVersion: cert-manager.io/v1
+   kind: Certificate
+   metadata:
+     name: envoy-gateway-ca
+     namespace: envoy-gateway-system
+   spec:
+     isCA: true
+     commonName: envoy-gateway
+     secretName: envoy-gateway-ca
+     privateKey:
+       algorithm: RSA
+       size: 2048
+     issuerRef:
+       name: selfsigned-issuer
+       kind: Issuer
+       group: cert-manager.io
+   ---
+   apiVersion: cert-manager.io/v1
+   kind: Issuer
+   metadata:
+     labels:
+       app.kubernetes.io/name: envoy-gateway
+     name: eg-issuer
+     namespace: envoy-gateway-system
+   spec:
+     ca:
+       secretName: envoy-gateway-ca
+   EOF
+   ```
+
+2. Create a cert for envoy gateway controller, the cert will be stored in secret `envoy-gatewy`.
+
+   ```shell
+   cat<<EOF | kubectl apply -f -
+   apiVersion: cert-manager.io/v1
+   kind: Certificate
+   metadata:
+     labels:
+       app.kubernetes.io/name: envoy-gateway
+     name: envoy-gateway
+     namespace: envoy-gateway-system
+   spec:
+     commonName: envoy-gateway
+     dnsNames:
+     - "envoy-gateway"
+     - "envoy-gateway.envoy-gateway-system"
+     - "envoy-gateway.envoy-gateway-system.svc"
+     - "envoy-gateway.envoy-gateway-system.svc.cluster.local"
+     issuerRef:
+       kind: Issuer
+       name: eg-issuer
+     usages:
+     - "digital signature"
+     - "data encipherment"
+     - "key encipherment"
+     - "content commitment"
+     secretName: envoy-gateway
+   EOF
+   ```
+
+3. Create a cert for envoy proxy, the cert will be stored in secret `envoy`.
+
+   ```shell
+   cat<<EOF | kubectl apply -f -
+   apiVersion: cert-manager.io/v1
+   kind: Certificate
+   metadata:
+     labels:
+       app.kubernetes.io/name: envoy-gateway
+     name: envoy
+     namespace: envoy-gateway-system
+   spec:
+     commonName: "*"
+     dnsNames:
+     - "*.envoy-gateway-system"
+     issuerRef:
+       kind: Issuer
+       name: eg-issuer
+     usages:
+     - "digital signature"
+     - "data encipherment"
+     - "key encipherment"
+     - "content commitment"
+     secretName: envoy
+   EOF
+   ```
+
+4. Create a cert for rate limit, the cert will be stored in secret `envoy-rate-limit`.
+
+   ```shell
+   cat<<EOF | kubectl apply -f -
+   apiVersion: cert-manager.io/v1
+   kind: Certificate
+   metadata:
+     labels:
+       app.kubernetes.io/name: envoy-gateway
+     name: envoy-rate-limit
+     namespace: envoy-gateway-system
+   spec:
+     commonName: "*"
+     dnsNames:
+     - "*.envoy-gateway-system"
+     issuerRef:
+       kind: Issuer
+       name: eg-issuer
+     usages:
+     - "digital signature"
+     - "data encipherment"
+     - "key encipherment"
+     - "content commitment"
+     secretName: envoy-rate-limit
+   EOF
+   ```
+
```

Comment:
should we add a note here about hmac-token-secret ? or not required ? cc @zhaohuabing

Reply:
TBH, I know little about that, maybe @zhaohuabing can update this later.

Reply:
HMAC Secret is not a cert and it can't be created by a cert manager.  Let's leave it for the certGen job until we add the doc for rotating the certs.